inc/integrations/class-form-settings-data.php

```diff
@@ -0,0 +1,483 @@
+	 *
+	 * @param array $integration_data The integration data.
+	 */
+	public function __construct( $integration_data ) {
+
+		$this->extract_integration_data( $integration_data );
+		if ( isset( $integration_data['hasCaptcha'] ) ) {
+			$this->set_captcha( $integration_data['hasCaptcha'] );
+		}
+
+		$this->set_meta( $integration_data );
+	}
+
+	/**
+	 * Extract the settings from 3rd party integration.
+	 *
+	 * @param array $integration_data The integration data.
+	 * @return void
+	 * @since 2.0.3
+	 */
+	public function extract_integration_data( $integration_data ) {
+		if ( isset( $integration_data['apiKey'] ) ) {
+			$this->set_api_key( $integration_data['apiKey'] );
+		}
+
+		if ( isset( $integration_data['listId'] ) ) {
+			$this->set_list_id( $integration_data['listId'] );
+		}
+
+		if ( isset( $integration_data['provider'] ) ) {
+			$this->set_provider( $integration_data['provider'] );
+		}
+
+		if ( isset( $integration_data['action'] ) ) {
+			$this->set_action( $integration_data['action'] );
+		}
+
+		if ( isset( $integration_data['listId'] ) ) {
+			$this->set_list_id( $integration_data['listId'] );
+		}
+	}
+
+	/**
+	 * Check if it has the necessary data set.
+	 *
+	 * @return string[] The issues about the missing settings.
+	 * @since 2.0.3
+	 */
+	public function check_data() {
+		$issues = array();
+
+		if ( ! $this->has_provider() ) {
+			$issues[] = __( 'Provider settings are missing!', 'otter-blocks' );
+		}
+
+		if ( ! $this->has_api_key() ) {
+			$issues[] = __( 'API Key is missing!', 'otter-blocks' );
+		}
+
+		if ( ! $this->has_list_id() ) {
+			$issues[] = __( 'Mail list is missing!', 'otter-blocks' );
+		}
+
+		return $issues;
+	}
+
+	/**
+	 * Get the 3rd party integration settings from WP options given the form option ID.
+	 *
+	 * @param string $form_option The ID of the form.
+	 * @return Form_Settings_Data
+	 * @since 2.0.3
+	 */
+	public static function get_form_setting_from_wordpress_options( $form_option ) {
+		$option_name = sanitize_text_field( $form_option );
+		$form_emails = get_option( 'themeisle_blocks_form_emails' );
+		$integration = new Form_Settings_Data( array() );
+		foreach ( $form_emails as $form ) {
+			if ( $form['form'] === $option_name ) {
+
+				if ( isset( $form['hasCaptcha'] ) ) {
+					$integration->set_captcha( $form['hasCaptcha'] );
+				}
+				if ( isset( $form['redirectLink'] ) ) {
+					$integration->set_redirect_link( $form['redirectLink'] );
+				}
+				if ( isset( $form['emailSubject'] ) ) {
+					$integration->set_email_subject( $form['emailSubject'] );
+				}
+				if ( isset( $form['submitMessage'] ) ) {
+					$integration->set_submit_message( $form['submitMessage'] );
+				}
+				if ( isset( $form['fromName'] ) ) {
+					$integration->set_from_name( $form['fromName'] );
+				}
+				if ( isset( $form['integration'] ) ) {
+					$integration->extract_integration_data( $form['integration'] );
+				}
+				$integration->set_meta( $form );
+			}
+		}
+		return $integration;
+	}
```

Comment:
Was there any reason why we were using options instead of post meta here? Don't think it's an issue for now but if we keep adding many such options then might be better to use a meta field instead. Or was it because Widgets didn't support meta? I can't recall the details so just wanted to know.

Reply:
I did not think about the post meta. I just reused this because it is already working. Also, I need to see if post meta is working with template parts from FSE.

Reply:
It's fine for now. You can check that later, maybe with a simple block and see if that is working. @Soare-Robert-Daniel